graphistry/pygraphistry.py

```diff
@@ -2446,7 +2461,140 @@ def _handle_api_response(response):
             logger.error('Error: %s', response, exc_info=True)
             raise Exception("Unknown Error")
 
+    @staticmethod
+    def sso_repeat_get_token(repeat: int = 20, wait: int = 5):
+        """ Function to repeatly call to obtain the jwt token after sso login
+        Function to obtain the JWT token after SSO login
+        """
+        
+        for _ in range(repeat):
+            token = PyGraphistry.sso_get_token()
+            if token:
+                return token
+            time.sleep(wait)
+
+        return
+
+    @staticmethod
+    def sso_wait_for_token_display(repeat: int = 20, wait: int = 5, fail_silent: bool = False, display_mode: str = 'text'):
+        if display_mode == 'html':
+            PyGraphistry.sso_wait_for_token_html_display(repeat, wait, fail_silent)
+        else:
+            PyGraphistry.sso_wait_for_token_text_display(repeat, wait, fail_silent)
+
+    @staticmethod
+    def sso_wait_for_token_text_display(repeat: int = 20, wait: int = 5, fail_silent: bool = False):
+        """Get the JWT token for SSO login and display corresponding message in text
+        Get the JWT token for SSO login and display corresponding message in text
+        """
+        if not PyGraphistry.api_token():
+            msg_text = '....'
+            if not PyGraphistry.sso_repeat_get_token(repeat, wait):
+                msg_text = f'{msg_text}\nFailed to get token after {repeat*wait} seconds ....'
```

Comment:
`msg_text = f'{msg_text}\nUnable to retrieve token after {repeat*wait} seconds. Please try logging in again.'
`

Reply:
done